test/dag_test.go

```diff
@@ -62,6 +62,12 @@ func TestDAGPipelineRun(t *testing.T) {
 		t.Fatalf("Failed to create echo Task: %s", err)
 	}
 
+	// Make sure the Pipeline has been created (wait for it)
+	if err := WaitForTaskCreated(c, "echo-task", "TaskCreated"); err != nil {
```

Comment:
Would it be worth adding a helper method to create-and-wait-for-existence a resource? I'm worried we'll forget to wait for existence every time we create a Task/Pipeline/etc., and have to chase flakes forever as a result.

Do we know why `Create` returns before the resource is actually reliably created? That seems like a problem for lots of tests. Is there somewhere we're using a lister in the reconciler to retrieve a Task/Pipeline, where we should be using a direct client to get them, bypassing caching? We've seen listers being slow to notice new objects in the past, and have (mostly?) replaced them with regular k8s clients as a result.

Reply:
I am not so sure and more I read your comment and more I think that my patch may not address the fundamental problem, it doesn't matter if we know it has been created if it's actually the reconciler being slow to be notified.

> Is there somewhere we're using a lister in the reconciler to retrieve a

I am not so sure about it (and probably will start to ping you about if i have to figure out :-)

Reply:
https://github.com/tektoncd/pipeline/blob/f1bfda69a04ae3bb159b9c8669136eab2b78f3b5/pkg/reconciler/pipelinerun/controller.go#L72-L82 ?